### DISCUSSION TITLE
Custom Data Source in extensibility library

### DISCUSSION THREAD

**Discussion by patrikhellgren** [microsoft-search/pnp-modern-search#2397] [Ideas]
I know this question has come up before (see #1471 and #2054) but without a real resolution so I am raising this again.
We were looking for a way to use a custom data source in Modern Search and found the mentioned threads and of course also noted that this feature is available in the commercial version.

I started looking into implementing this in our own fork and now have a working version with custom data source extensibility so the question is if this is of interest to this project? If it is I can submit a PR and since there is only a few lines of changes it would not be too much work to review it. You can also find the changes I have made [here](https://github.com/SherpasGroup/pnp-modern-search/commit/551dc49880492575b457a4255e1487d13b7d5088).

I guess documentation and examples would also be needed.

What are your thoughts?

**Comment by wobba** [comment]
@patrikhellgren I'd like to review this and having a dummy/sample provider as part of https://github.com/microsoft-search/pnp-modern-search-extensibility-samples would be useful + docs yes :)

**Reply by FranckyC** [reply]
Hi @patrikhellgren, sure! Feel free to do PR and we will review it. In this case, please also open a PR in the https://github.com/microsoft-search/pnp-modern-search-extensibility-samples repo with an example data source. Thanks!

**Reply by patrikhellgren** [reply]
@wobba and @FranckyC , I have now submitted two PRs for this, #2400 and microsoft-search/pnp-modern-search-extensibility-samples#11.

**Comment by ggravey** [comment]
Hello @wobba @FranckyC.

I am really interested in this new feature. (Thank you @patrikhellgren)
Is there any chance for this change to be included in V4.8 ?

Thank you
